_posts/2021-08-30-changelog-154.markdown

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #154"
+date: 2021-08-30 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#7640] Improve [external ABI](https://doc.rust-lang.org/reference/items/external-blocks.html#abi) support:
+    - Parse any literal in ABI name position (including raw strings) and annotate non-string literals as invalid (in the same way as `rustc` does)
+
+      <img src="/assets/posts/changelog-154/non-string-abi-literal.png" alt="Non-string abi literal" width="700px">
+    - Annotate [invalid ABI](https://doc.rust-lang.org/error-index.html#E0703) names and experimental ABIs without the corresponding feature attribute
+
+      <img src="/assets/posts/changelog-154/invalid-abi-name.png" alt="Invalid abi name" width="700px">
+    - Provide competition for ABI names
+
+      {% include gif-img.html path="/assets/posts/changelog-154/abi-completion" w="700px" %}
+
+* [#7682] Detect [E0121](https://doc.rust-lang.org/error-index.html#E0121) in structs and enum variants (by [@t-kameyama])
+
+  <img src="/assets/posts/changelog-154/type-placeholder.png" alt="The type placeholder `_` is not allowed within types on item signatures for structs and enum variants" width="700px">
+
+* [#7646] Implement `Expand into separate table` intention to expand an inline TOML table into a full one
+
+  {% include gif-img.html path="/assets/posts/changelog-154/expand-into-separate-table" w="700px" %}
+
+* [#7666] Show [macro 2.0](https://rust-lang.github.io/rfcs/1584-macros.html) items in [Structure View](https://www.jetbrains.com/help/idea/viewing-structure-of-a-source-file.html) (`Ctrl`+`F12`)
+
+* [#7664] Improve parser recovery for [function pointers](https://doc.rust-lang.org/reference/types/function-pointer.html) (by [@afetisov])
+
+## Performance Improvements
+
+* [#7703] Suggest keywords much more quickly during Rust code completion
+
+## Fixes
+
+[comment]: <> (* [#7711] Introduce `Ty.isEquivalentTo&#40;Ty&#41;` for `Ty`s comparison. Unlike `== `, it ignores `aliasedBy`)
+[comment]: <> ( field.)
+* [#7635], [#7676] Take into account type aliases in most of [intention actions, quick fixes](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-generation.html#intention-actions), and [refactorings](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html) (by [@Kobzol])
+
+* [#7557] Reduce amount of false positive annotations caused by `Self convention` inspection (by [@Kobzol])
+
+* [#7692] Don't wrap associated types by type aliases in `Chained method` inlay hints (by [@Kobzol])
```

Comment:
In theory, this affects all inlay type hints, not only `Chained method` hints. But it's true that this is most useful for `Chained method` hints